### PR TITLE
Library new project actions and fixes

### PR DESCRIPTION
**Pull request by paulovmr**
GUVNOR-2919: Library: Add Build, Deploy and Messages actions to the project screen
GUVNOR-2965: [Library] Breadcrumb navigation to LibraryView causes errors
GUVNOR-2966: [Library] After creation of a new project, other project is open

Includes a refactoring to remove the build logic from `ProjectScreenPresenter` to the new `BuildExecutor`.

### CODE REVIEW

File: kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/java/org/kie/workbench/common/screens/library/client/util/LibraryPlaces.java [hunk 1]
```diff
@@ -72,6 +72,7 @@
     public static final String PROJECT_DETAIL_SCREEN = "ProjectsDetailScreen";
     public static final String PROJECT_SETTINGS = "projectScreen";
     public static final String PROJECT_EXPLORER = "org.kie.guvnor.explorer";
+    public static final String MESSAGES = "org.kie.workbench.common.screens.messageconsole.MessageConsole";
```
Comment (jhrcek) [inline kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/java/org/kie/workbench/common/screens/library/client/util/LibraryPlaces.java]: There is an annoying glitch with messages panel: whenever it's displayed in the library, there's a white margin above the panel. Could you please review the CSS and remove it?
annoyingmargin

File: kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/java/org/kie/workbench/common/screens/library/client/screens/ProjectView.java [hunk 2]
```diff
@@ -75,41 +83,51 @@
...
-    public void init( ProjectScreen presenter ) {
+    public void init(ProjectScreen presenter) {
         this.presenter = presenter;
-        projectActionsWidget.init( () -> presenter.goToSettings() );
-        filterText.setAttribute( "placeholder", ts.getTranslation( LibraryConstants.LibraryView_Filter ) );
-        detailsContainer.appendChild( projectsDetailScreen.getView().getElement() );
-        toolbar.appendChild( projectActionsWidget.getView().getElement() );
+        assetsActionsWidget.init();
+        projectActionsWidget.init(presenter::goToSettings);
+        filterText.setAttribute("placeholder",
```
Comment (paulovmr) [inline kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/java/org/kie/workbench/common/screens/library/client/screens/ProjectView.java]: I couldn't do this on the template last time I tried... 

@mbarkley Hi, do you know if there is a way to do that?

File: kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/java/org/kie/workbench/common/screens/library/client/util/LibraryPlaces.java (hunk 1 above)
Comment (paulovmr) [inline kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/java/org/kie/workbench/common/screens/library/client/util/LibraryPlaces.java]: @jhrcek Could you open another JIRA issue for that and assign it to me (since the problem is not related to these PR changes)?

Comment (jhrcek) [inline kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/java/org/kie/workbench/common/screens/library/client/util/LibraryPlaces.java]: Sure, will do.

File: kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/java/org/kie/workbench/common/screens/library/client/widgets/ProjectActionsView.html [hunk 3]
```diff
@@ -15,13 +15,16 @@
...
-    <div class="dropdown btn-group dropdown-kebab-pf">
-        <button type="button" class="btn btn-primary dropdown-toggle" data-toggle="dropdown" id="dropdownKebab"
-                aria-haspopup="true" aria-expanded="true" data-i18n-key="CreateNewAsset"></button>
-        <ul data-field="resource-handler-menu-container" class="large dropdown-menu dropdown-menu-right"
-            aria-labelledby="dropdownKebab"></ul>
-    </div>
     <div class="btn-group">
         <button type="button" class="btn" data-field="settings" data-i18n-key="Settings"></button>
     </div>
+    <div class="btn-group">
```
Comment (Rikkola) [inline kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/java/org/kie/workbench/common/screens/library/client/widgets/ProjectActionsView.html]: I would expect these to be removed from Project Editor and Project Editor is used 100% just for configuring.

But I do admit that the task was just to add them, not to move them. So should be checked what is expected.

File: kie-wb-common-screens/kie-wb-common-project-editor/kie-wb-common-project-editor-client/src/main/java/org/kie/workbench/common/screens/projecteditor/client/build/BuildExecutor.java [hunk 4]
```diff
@@ -0,0 +1,369 @@
...
+
+    public void triggerBuild() {
+        getSafeExecutedCommand(getBuildCommand()).execute();
+    }
+
+    public void triggerBuildAndDeploy() {
+        specManagementService.call(new RemoteCallback<Collection<ServerTemplate>>() {
+            @Override
+            public void callback(final Collection<ServerTemplate> serverTemplates) {
+                //TODO use config for using defaults or not
```
Comment (Rikkola) [inline kie-wb-common-screens/kie-wb-common-project-editor/kie-wb-common-project-editor-client/src/main/java/org/kie/workbench/common/screens/projecteditor/client/build/BuildExecutor.java]: Better to make a Jira ticket for the TODO than send it to master. Leave it here and nobody dares to touch the codes here.

File: kie-wb-common-screens/kie-wb-common-project-editor/kie-wb-common-project-editor-client/src/main/java/org/kie/workbench/common/screens/projecteditor/client/build/BuildExecutor.java [hunk 5]
```diff
@@ -0,0 +1,369 @@
...
+                        getSafeExecutedCommand(getBuildDeployProvisionCommand(DeploymentMode.VALIDATED,
+                                                                              containerId,
+                                                                              containerAlias,
+                                                                              serverTemplate,
+                                                                              startContainer)).execute();
+
+                        deploymentScreenPopupView.hide();
+                    });
+                    deploymentScreenPopupView.show();
+                }
```
Comment (Rikkola) [inline kie-wb-common-screens/kie-wb-common-project-editor/kie-wb-common-project-editor-client/src/main/java/org/kie/workbench/common/screens/projecteditor/client/build/BuildExecutor.java]: Would be nice to see these chunks of code in if/else brackets placed into methods.

File: kie-wb-common-screens/kie-wb-common-project-editor/kie-wb-common-project-editor-client/src/main/java/org/kie/workbench/common/screens/projecteditor/client/editor/ProjectScreenPresenter.java [hunk 6]
```diff
@@ -818,354 +777,152 @@ public void execute() {
...
-                                                    final String serverTemplate,
-                                                    final boolean startContainer ) {
-        return new Command() {
-            @Override
-            public void execute() {
-                view.showBusyIndicator( ProjectEditorResources.CONSTANTS.Building() );
-                buildAndDeployAndProvision( mode, containerId, containerAlias, serverTemplate, startContainer );
-            }
-        };
+        runBuild(() -> buildExecutor.triggerBuildAndDeploy());
```
Comment (Rikkola) [inline kie-wb-common-screens/kie-wb-common-project-editor/kie-wb-common-project-editor-client/src/main/java/org/kie/workbench/common/screens/projecteditor/client/editor/ProjectScreenPresenter.java]: Good to see this class get split. It is getting big. I see the comments I made above were about old code. Still best to kill TODOs and split code blocks to methods when we run into them.

File: kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/java/org/kie/workbench/common/screens/library/client/widgets/ProjectActionsView.html (hunk 3 above)
Comment (paulovmr) [inline kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/java/org/kie/workbench/common/screens/library/client/widgets/ProjectActionsView.html]: @jhrcek They will be available in both places. I applied the same style now, just the size will be different.

File: kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/java/org/kie/workbench/common/screens/library/client/screens/ProjectView.java (hunk 2 above)
Comment (mbarkley) [inline kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/java/org/kie/workbench/common/screens/library/client/screens/ProjectView.java]: @paulovmr You can translate element attributes declaratively with Errai i18n. See the comment [here](https://issues.jboss.org/browse/ERRAI-962).

**Comment by manstis** [comment]
Jenkins please retest this.

File: kie-wb-common-screens/kie-wb-common-project-editor/kie-wb-common-project-editor-client/src/main/java/org/kie/workbench/common/screens/projecteditor/client/build/BuildExecutor.java (hunk 4 above)
Comment (paulovmr) [inline kie-wb-common-screens/kie-wb-common-project-editor/kie-wb-common-project-editor-client/src/main/java/org/kie/workbench/common/screens/projecteditor/client/build/BuildExecutor.java]: Done: https://issues.jboss.org/browse/GUVNOR-2969

File: kie-wb-common-screens/kie-wb-common-project-editor/kie-wb-common-project-editor-client/src/main/java/org/kie/workbench/common/screens/projecteditor/client/build/BuildExecutor.java (hunk 5 above)
Comment (paulovmr) [inline kie-wb-common-screens/kie-wb-common-project-editor/kie-wb-common-project-editor-client/src/main/java/org/kie/workbench/common/screens/projecteditor/client/build/BuildExecutor.java]: Done.

File: kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/java/org/kie/workbench/common/screens/library/client/screens/ProjectView.java (hunk 2 above)
Comment (paulovmr) [inline kie-wb-common-screens/kie-wb-common-library/kie-wb-common-library-client/src/main/java/org/kie/workbench/common/screens/library/client/screens/ProjectView.java]: @jhrcek Since that is not related to this PR (the line was just formatted), could you please open another ticket for that (if you feel necessary)? Thanks.
